src/mappings/Rewards.ts

```diff
@@ -70,6 +68,27 @@ async function handleRewardForTxHistory(rewardEvent: SubstrateEvent): Promise<vo
     )
 }
 
+function determinePayoutCallsArgs(causeCall: CallBase<AnyTuple>) : [string, number][] {
+    if (isPayoutStakers(causeCall)) {
+        return [extractArgsFromPayoutStakers(causeCall)]
+    } else if (isBatch(causeCall)) {
+        let calls = callsFromBatch(causeCall)
+        let payoutStakers = calls
+            .filter(isPayoutStakers)
+            .map(extractArgsFromPayoutStakers)
+
+        let proxyCalls = calls
+            .filter(isProxy)
+            .map(callFromProxy)
+            .flatMap(determinePayoutCallsArgs)
+
+        return payoutStakers.concat(proxyCalls)
```

Comment:
This approach breaks original order of calls which is important in our algo that determines a validator per event